commands/checkout_test.go

```diff
@@ -0,0 +1,16 @@
+package commands
+
+import (
+	"testing"
+
+	"github.com/github/hub/Godeps/_workspace/src/github.com/bmizerany/assert"
+)
+
+func TestReplaceCheckoutParam(t *testing.T) {
+	checkoutURL := "https://github.com/github/hub/pull/12"
+	args := NewArgs([]string{"checkout", "-b", checkoutURL})
+	replaceCheckoutParam(args, checkoutURL, "jingweno", "origin/master")
+
+	cmd := args.ToCmd()
+	assert.Equal(t, "git checkout -b --track -B jingweno origin/master", cmd.String())
```

Comment:
We could do a check of `-b` before everything else, currently it fails on the `git-checkout`:

```
$ ./hub checkout -b https://github.com/github/hub/pull/813
Updating github
From https://github.com/github/hub
 * [new branch]      relax_insert_param_index -> github/relax_insert_param_index
fatal: -b, -B and --orphan are mutually exclusive
```

I would still want to keep the changes of `Args.InsertParam` though since it's more resilient to exceptions.